1-org/modules/centralized-logging/README.md

```diff
@@ -73,7 +75,7 @@ module "logging_logbucket" {
-| logbucket\_options | Destination LogBucket options:<br>- name: The name of the log bucket to be created and used for log entries matching the filter.<br>- logging\_sink\_name: The name of the log sink to be created.<br>- logging\_sink\_filter: The filter to apply when exporting logs. Only log entries that match the filter are exported. Default is "" which exports all logs.<br>- location: The location of the log bucket. Default: global.<br>- retention\_days: The number of days data should be retained for the log bucket. Default 30. | <pre>object({<br>    name                = optional(string, null)<br>    logging_sink_name   = optional(string, null)<br>    logging_sink_filter = optional(string, "")<br>    location            = optional(string, "global")<br>    retention_days      = optional(number, 30)<br>  })</pre> | `null` | no |
+| logbucket\_options | Destination LogBucket options:<br>- name: The name of the log bucket to be created and used for log entries matching the filter.<br>- logging\_sink\_name: The name of the log sink to be created.<br>- logging\_sink\_filter: The filter to apply when exporting logs. Only log entries that match the filter are exported. Default is "" which exports all logs.<br>- location: The location of the log bucket. Default: global.<br>- enable\_analytics: Whether or not Log Analytics is enabled. A Log bucket with Log Analytics enabled can be queried in the Log Analytics page using SQL queries. Cannot be disabled once enabled.<br>- linked\_dataset\_id: The ID of the linked BigQuery dataset. A valid link dataset ID must only have alphanumeric characters and underscores within it and have up to 100 characters.<br>- linked\_dataset\_description: A use-friendly description of the linked BigQuery dataset. The maximum length of the description is 8000 characters.<br>- retention\_days: The number of days data should be retained for the log bucket. Default 30. | <pre>object({<br>    name                       = optional(string, null)<br>    logging_sink_name          = optional(string, null)<br>    logging_sink_filter        = optional(string, "")<br>    location                   = optional(string, "global")<br>    enable_analytics           = optional(bool, true)<br>    linked_dataset_id          = optional(string, null)<br>    linked_dataset_description = optional(string, null)<br>    retention_days             = optional(number, 30)<br>  })</pre> | `null` | no |
```

Comment:
Double checking... does linked_dataset_id allow you to specify a name for the linked dataset different from the name of the log bucket? 

I thought the linked dataset ID always took the name of the bucket because the gcloud command line doesn't have an option to specify a new dataset_id.
https://cloud.google.com/logging/docs/buckets#link-bq-dataset

(If this is just an inconsistency between the API and gcloud, then it's not a blocking issue)

Reply:
In Terraform the link id will be the name of the BigQuery dataset, I am not familiar with the gcloud command usage for creating the link

checking  In the logging API we have:

[Log bucket create](https://cloud.google.com/logging/docs/reference/v2/rest/v2/projects.locations.buckets.links/create#query-parameters) - **linkId**


```
Required. The ID to use for the link. The linkId can have up to 100 characters. A valid linkId must only have alphanumeric characters and underscores within it.
```


[Link.BigQueryDataset](https://cloud.google.com/logging/docs/reference/v2/rest/v2/billingAccounts.locations.buckets.links#Link.BigQueryDataset) - **datasetId**

```
Output only. The full resource name of the BigQuery dataset. The DATASET_ID will match the ID of the link, so the link must match the naming restrictions of BigQuery datasets (alphanumeric characters and underscores only).

The dataset will have a resource path of "bigquery.googleapis.com/projects/[PROJECT_ID]/datasets/[DATASET_ID]"
```

it is the same as in the [API version](https://cloud.google.com/logging/docs/buckets#api_3) of the instructions for creation of the linked dataset.

Maybe the documentation for the usage of the `gcloud` command needs to be fixed?